app/routes/questionnaire.py

```diff
@@ -227,9 +227,10 @@ def block(schema, questionnaire_store, block_id, list_name=None, list_item_id=No
         submission_handler.submit_questionnaire()
 
         language_code = get_session_store().session_data.language_code
-        log_out_url = get_census_base_url(cookie_session["theme"], language_code)
-
-        cookie_session["account_service_log_out_url"] = log_out_url
+        if "census" in (
+            log_out_url := get_census_base_url(cookie_session["theme"], language_code)
+        ):
+            cookie_session["account_service_log_out_url"] = log_out_url
```

Comment:
Umm if you are adding census specific code then why not just check the theme is a census theme. Instead of checking if the phrase 'census' is in the thank you URL?

Reply:
Also 'census' wouldn't be in the URL when the language code is Welsh.

Reply:
whoops I've made an error here - I wanted to test against the cookie_session theme not the base url